src/components/portal/portal-arrow.scss

```diff
@@ -0,0 +1,94 @@
+/**
+ * @prop --portal-arrow-color: The color of the arrow has to be the same as the surface opened in the portal.
+ */
+
+$size-of-arrow: 0.5rem;
+
+limel-portal-arrow,
+limel-portal-arrow::before {
+    position: absolute;
+}
+
+limel-portal-arrow {
+    visibility: hidden;
+
+    &::before {
+        visibility: visible;
+        content: '';
+
+        border: {
+            width: $size-of-arrow;
+            color: transparent;
+            style: solid;
+        }
+        [data-popper-placement^='top'] & {
+            border-top-color: var(--portal-arrow-color);
+            margin-left: -$size-of-arrow;
+        }
+        [data-popper-placement^='bottom'] & {
+            border-bottom-color: var(--portal-arrow-color);
+            margin-left: -$size-of-arrow;
+        }
+        [data-popper-placement^='left'] & {
+            border-left-color: var(--portal-arrow-color);
+            margin-top: -$size-of-arrow;
+        }
+        [data-popper-placement^='right'] & {
+            border-right-color: var(--portal-arrow-color);
+            margin-top: -$size-of-arrow;
+        }
+    }
+
+    [data-popper-placement^='top'] & {
+        bottom: 0;
+    }
+    [data-popper-placement^='bottom'] & {
+        top: -$size-of-arrow * 2;
+    }
+    [data-popper-placement^='left'] & {
+        right: 0;
+    }
+    [data-popper-placement^='right'] & {
+        left: -$size-of-arrow * 2;
+    }
+}
+
+.limel-portal--container {
+    &.has-arrow {
+        &[data-popper-placement^='right'] {
+            margin-left: $size-of-arrow !important;
+        }
+        &[data-popper-placement^='left'] {
+            margin-right: $size-of-arrow !important;
+        }
+        &[data-popper-placement^='top'] {
+            margin-bottom: $size-of-arrow !important;
+        }
+        &[data-popper-placement^='bottom'] {
+            margin-top: $size-of-arrow !important;
+        }
+    }
+}
+
+.arrow-style-tooltip::before {
+    --portal-arrow-color: rgb(
+        var(--contrast-1300)
+    ); // the same color as the tooltip background-color
+    opacity: 0;
+    animation: display-tooltip 0.2s ease forwards;
+
+    animation-delay: 0.2s;
+}
+
+.arrow-style-popover::before {
+    --portal-arrow-color: var(--popover-body-background-color);
```

Comment:
Popover is a consumer of portal, so portal should not have to know anything about popover. Popover is the one that should make this "conversion" from `--popover-body-background-color` to some other variable prefixed with `--portal-`.

Reply:
Sounds OK for me. Something like `--portal-surface-background-color`

Reply:
No, what I mean is that limel-portal shouldn't have to be aware of css variables that are "owned" by other components. If we have a `--popover-z-index`, a `--tooltip-z-index`, and a `--menu-z-index`, limel-portal should not need to be aware of any of those. Instead, limel-popover, limel-tooltip, and limel-menu, respectively, should take care of setting a variable called something like `--portal-z-index`, using the value from their own respective variable. (Also, we shouldn't just cut a corner tell the consumer to use `--portal-z-index` directly, because the consumer shouldn't have to know or care that limel-popover, limel-tooltip, or limel-menu is using limel-portal internally.)

Reply:
This was easy to fix for popover, but…